cluster-up/cluster/kind/common.sh

```diff
@@ -210,9 +196,91 @@ function setup_kind() {
     prepare_config
 }
 
+function ensure_memory_dir_for_etcd_data() {
+    if [ "$ETCD_DATA_DIR" == none ];then
+        echo "ETCD_DATA_DIR is not defined" && exit 1
+    fi
+
+    if grep -qs $ETCD_DATA_DIR /proc/mounts; then
+        echo "etcd data mount already exists"
+        df -h $ETCD_DATA_DIR
+        if [ -d "$ETCD_DATA_DIR" ];then
+            echo "ectd data directory already exists, cleaning up directory content..."
+            rm -rf $ETCD_DATA_DIR/*
+        fi
+    else
+        echo "creating directory for ectd data"
+        mkdir -p $ETCD_DATA_DIR
+        [ "echo $?" != 0 ] && echo "failed to create etcd data directory" && exit 1
+        ls -lsah $ETCD_DATA_DIR
+
+        echo "mount etcd data directory as tempfs"
+        mount -t tmpfs tmpfs $ETCD_DATA_DIR
```

Comment:
Maybe I am wrong but when we run this on prow the tmpfs is done by k8s when we crate the volume of typoe `Memory`, right, so we don't need to do this ourselves.

Reply:
Great question :) 
I added this logic to support development environments flow where you dont have k8s to 
create this mount for you.
On local dev env this function will create a directory and mount it, 
if the mount already exists it will delete the mount content to prevent contradicts with new data.
On CI since we ask from the k8s to create a mount for us,
this function will check that the mount exists, delete the mount content to prevent contradicts with new data
and prints its details

Reply:
at local env we have to also remove it after we do cluster-down right ?

Reply:
You right I will add a clean up for this on cluster-down